lib/src/hook_file.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:bdd_widget_test/src/generator_options.dart';
+import 'package:bdd_widget_test/src/util/constants.dart';
+import 'package:path/path.dart' as p;
+
+class HookFile {
+  HookFile._create({
+    required this.featureDir,
+    required this.package,
+    required this.fileName,
+    required this.import,
+  });
+
+  static HookFile create({
+    required String featureDir,
+    required String package,
+    required GeneratorOptions generatorOptions,
+  }) {
+    const fileName = '$hookFileName.dart';
+    if (generatorOptions.hookFolderName.startsWith('./') ||
+        generatorOptions.hookFolderName.startsWith('../')) {
+      return HookFile._create(
+        featureDir: featureDir,
+        package: package,
+        fileName: p.join(featureDir, generatorOptions.hookFolderName, fileName),
+        import: p
+            .join(generatorOptions.hookFolderName, fileName)
+            .replaceAll(r'\', '/'),
+      );
+    }
+
+    return HookFile._create(
+      featureDir: featureDir,
+      package: package,
+      fileName:
+          p.join(testFolderName, generatorOptions.hookFolderName, fileName),
+      import: p.join(
+        p.relative(testFolderName, from: featureDir),
+        generatorOptions.hookFolderName,
+        fileName,
+      ),
+    );
+  }
+
+  final String featureDir;
+  final String fileName;
+  final String package;
+  final String import;
+}
+
+const String hookFileContent = '''
+class Hooks {
+  static Function beforeEach = (String title, [List<String>? tags]) {
+    // Add logic for beforeEach
+  };
+  static Function beforeAll = () {
+    // Add logic for beforeAll
+  };
+  static Function afterEach = (
+    String title,
+    bool success, [
+    List<String>? tags,
+  ]) {
+    // Add logic for afterEach
+  };
+  static Function afterAll = () {
+    // Add logic for afterAll
+  };
+}
```

Comment:
I guess class is highly optional, but if you think it is needed, please make it abstract with private a constructor. It is better to replace `dynamic`s with proper `Future<void>` or `FutureOr<void>` types, some users might have [this rule](https://dart.dev/tools/linter-rules/prefer_function_declarations_over_variables) enabled, hence it is better to declare functions as functions.

```suggestion
import 'dart:async';

abstract class Hooks {
  const Hooks._();

  static FutureOr<void> beforeEach(String title, [List<String>? tags]) {
    // Add logic for beforeEach
  }

  static FutureOr<void> beforeAll() {
    // Add logic for beforeAll
  }

  static FutureOr<void> afterEach(
    String title,
    bool success, [
    List<String>? tags,
  ]) {
    // Add logic for afterEach
  }

  static FutureOr<void> afterAll() {
    // Add logic for afterAll
  }
}
```